### DISCUSSION TITLE
Fuzzy Search

### DISCUSSION THREAD

**Discussion by pkranga** [Sunbird-Knowlg/Community#11] [General Discussions]
Does the current search API handle spell mistake search as well. for example if we search for "pulvariser" does it return content that has the word "pulvarizer" as well?

We tried using the fuzzy: true in the search request it just returns all the records. 

@vinukumar-vs

**Comment by Jayaprakash8887** [comment]
@pkranga No. fuzzy search is not implemented in the search API.

**Reply by pramodkvarma** [reply]
@pkranga maybe you can implement and contribute back, that'd be wonderful. If you are doing, use this thread to discuss design etc.

**Comment by Jayaprakash8887** [comment]
@pkranga sincere apologies. Seems like necessary coding for fuzzy search is available. I am gathering information on how to utilize it. will revert at the earliest.

**Reply by Jayaprakash8887** [reply]
@pkranga As you have correctly mentioned, current code seems to be returning all records when `"fuzzy": true`. I couldnt find a way to filter for necessary record using fuzzy capability. Looking forward for your contribution.

**Reply by vinukumar-vs** [reply]
@pkranga Fuzzy search is available out of the box by default from ES. We have not explored this. Please try
https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-fuzzy-query.html

@maheshkumargangula share your view if you have already explored this option.

**Reply by pkranga** [reply]
@vinukumar-vs @Jayaprakash8887 We have completed the fix for the fuzzy search. Let us know against which branch the PR has to be raised

**Reply by vinukumar-vs** [reply]
@pkranga Presently we are working on release-5.0.0. You can rise the PR for this branch(release-5.0.0).

For more details about release timelines please refer [Sunbird-Knowlg release dates]( https://knowlg.sunbird.org/use/release)
